VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsSearchFragment.java

```diff
@@ -106,7 +107,9 @@ protected void onException(Exception ex) {
                 nextPage = 1;
             }
         };
-        searchThreadListTask.setProgressCallback(null);
+        if (nextPage > 1 || callback.isRefreshingSilently()) {
```

Comment:
Maybe add a comment explaining the logic we're trying to implement here; i.e. show the central progress bar initially, the let the list view pagination handle it subsequently.